eth/db_upgrade.go

```diff
@@ -93,13 +93,19 @@ func upgradeSequentialKeys(db ethdb.Database) (stopFn func()) {
 func upgradeSequentialCanonicalNumbers(db ethdb.Database, stopFn func() bool) (error, bool) {
 	prefix := []byte("block-num-")
 	it := db.(*ethdb.LDBDatabase).NewIterator()
+	defer func() {
+		it.Release()
+	}()
```

Comment:
Or no, it's actually correct this way here because the value of `it` changes in the loop.